array/DNAm/analysis/EWAS/testCellComposition.r

```diff
@@ -0,0 +1,165 @@
+
+library(CETYGO)
+library(gridExtra)
+library(ggplot2)
+library(tidyr)
+library(dplyr)
+
+#----------------------------------------------------------------------#
+# DEFINE PARAMETERS
+#----------------------------------------------------------------------#
+
+args<-commandArgs(trailingOnly = TRUE)
+dataDir <- args[1]
+
+normData<-file.path(dataDir, "/3_normalised/normalised.rdata")
+resPath<-file.path(dataDir, "/4_analysis")
+
+#----------------------------------------------------------------------#
+# LOAD AND PREPARE DATA
+#----------------------------------------------------------------------#
+
+setwd(dataDir)
+load(normData)
+
+## remove total samples and cell types with less than 20 samples
+QCmetrics<-QCmetrics[which(QCmetrics$Cell.type != "Total"),]
+nSample<-table(QCmetrics$Cell.type)
+QCmetrics<-QCmetrics[QCmetrics$Cell.type %in% names(nSample[which(nSample > 19)]),]
+
+# filter to schizophrenia and control only
+QCmetrics<-QCmetrics[QCmetrics$Phenotype %in% c("Schizophrenia", "Control"),]
+
+celltypeNormbeta<-celltypeNormbeta[,QCmetrics$Basename]
+cellTypes<-unique(QCmetrics$Cell.type)
+
+predPropBest<-calcBrainComposition(celltypeNormbeta)
+
+save(predPropBest, file = file.path(resPath, "EstimatedNeuralCellComposition.rdata"))
+
+#----------------------------------------------------------------------#
+# PLOT DISTRIBUTIONS
+#----------------------------------------------------------------------#
+
+predPropBest<-cbind(QCmetrics, predPropBest)
+
+datLong<-gather(predPropBest[,c("Cell.type", "Phenotype", "NeuNNeg_SOX10Neg","NeuNPos","NeuNNeg_Sox10Neg_IRF8Pos","NeuNNeg_Sox10Neg_IRF8Neg","SATB2Neg","SATB2Pos","NeuNNeg","NeuNPos_SOX6Pos","NeuNPos_SOX6Neg","NeuNNeg_SOX10Pos", "CETYGO")], Fraction, Proportion, NeuNNeg_SOX10Neg:CETYGO)
+
+sumStats<-cbind(aggregate(Proportion ~ ., data = datLong, FUN = mean),
+aggregate(Proportion ~ ., data = datLong, FUN = sd)[,4])
+write.csv(sumStats, file.path(resPath, "EWAS", "Tables", "CellCompositionSummaryStatisticsByCellTypePhenotype.csv"))
+
+testPhenotype<-NULL
+for(fraction in c("NeuNNeg_SOX10Neg","NeuNPos","NeuNNeg_Sox10Neg_IRF8Pos","NeuNNeg_Sox10Neg_IRF8Neg","SATB2Neg","SATB2Pos","NeuNNeg","NeuNPos_SOX6Pos","NeuNPos_SOX6Neg","NeuNNeg_SOX10Pos", "CETYGO")){
+	for(sampletype in unique(predPropBest$Cell.type)){
+		subData<-subset(datLong, Cell.type == sampletype & Fraction == fraction)
+		testDat<-t.test(Proportion ~ Phenotype, data = subData)
+		testPhenotype<-rbind(testPhenotype, c(sampletype, fraction, testDat$estimate, testDat$p.value))
+	}
+
+}
+colnames(testPhenotype)<-c("Cell.type", "Fraction", "MeanControls", "MeanSchizophrenia", "Pvalue")
+write.csv(testPhenotype, file.path(resPath,  "EWAS", "Tables", "CellCompositionTtestStatisticsByCellTypePhenotype.csv"))
+
+# confirm purity of each cell type
+datLong<-gather(predPropBest[,c("Cell.type", "Phenotype", "NeuNPos", "NeuNNeg_SOX10Pos", "NeuNNeg_SOX10Neg")], Fraction, Proportion, NeuNPos:NeuNNeg_SOX10Neg)

```

Comment:
I feel like the rest of this file is quite repetitive. Is it possible to put these plotting sections into a function? On initial inspection each plotting section looks very similar, but maybe this isn't the case and converting to a function wouldn't clean the code up too much.

Reply:
You probably could but wasn't a priority at the time.